src/legacy/core_plugins/kibana/public/discover/saved_searches/_saved_search.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { SavedObjectKibanaServices } from 'ui/saved_objects/types';
+import { createSavedObjectClass } from 'ui/saved_objects/saved_object';
+
+export function createSavedSearchClass(services: SavedObjectKibanaServices) {
+  const SavedObjectClass = createSavedObjectClass(services);
+
+  class SavedSearch extends SavedObjectClass {
+    public static type: string = 'search';
+    public static mapping = {
+      title: 'text',
+      description: 'text',
+      hits: 'integer',
+      columns: 'keyword',
+      sort: 'keyword',
+      version: 'integer',
+    };
+    // Order these fields to the top, the rest are alphabetical
+    public static fieldOrder = ['title', 'description'];
+    public static searchSource = true;
+
+    public id: string;
```

Comment:
I adapted this change, since it makes sense, but had to revert,  because for some reason inheritance doesn't work in this case (Functional tests were failing). could be because we don't inherit directly but generate the Class to inherit from. A pattern we should and will change.